packages/trader/src/Modules/Account/Sections/Profile/PersonalDetails/personal-details.jsx

```diff
@@ -211,8 +210,16 @@ class PersonalDetailsForm extends React.Component {
             }
         }
 
-        if (values.address_postcode && !validPostCode(values.address_postcode)) {
-            errors.address_postcode = localize('Only letters, numbers, space, and hyphen are allowed.');
+        if (values.address_postcode) {
+            if (!validPostCode(values.address_postcode)) {
+                errors.address_postcode = localize('Only letters, numbers, space, and hyphen are allowed.');
+            }
+            if (!validLength(values.address_postcode, { min: 0, max: 20 })) {
```

Comment:
length check should come first, no?

Reply:
absolutely, thanks.

Reply:
@negarn I just remembered why I had done it that way. The validation goes on, and latest message is written to `errors.address_postcode`. So if length validation is first, then `validPostCode` will overwrite it if the length doesn't match in its validator function. So length should come second.